pkgs/by-name/je/jenkins/package.nix

```diff
@@ -16,13 +16,13 @@
   curl,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "jenkins";
-  version = "2.504.1";
+  version = "2.504.2";
```

Comment:
Does this get shadowed by `finalAttrs.version`?

Reply:
I mean to ask whether this value is unused since `finalAttrs.version` is used throughout the rest of the derivation.

Reply:
No, it's used.

Reply:
The finalAttrs pattern is a replacement for `rec`.
You can think of it like a loop: `finalAttrs = (inline lambda) finalAttrs` that is repeated until finalAttrs no longer changes. (Here, the inline lambda is everything after stdenv.mkDerivation.)